hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieFileIndex.scala

```diff
@@ -223,6 +223,21 @@ case class HoodieFileIndex(spark: SparkSession,
 
     val prunedPartitionsAndFileSlices = getFileSlicesForPrunedPartitions(partitionFilters)
 
+    val prunedPartitionFileNames: Set[String] = {
```

Comment:
Can we do this transformation inside `lookupCandidateFilesInMetadataTable`? Whereas `!isMetadataTableEnabled || !isDataSkippingEnabled` queries can avoid this

Reply:
done